### ISSUE TITLE
Huge FPS drop when a redstone engine is placed in the world

### ISSUE THREAD

**Issue by BBoldt** [AlexIIL/Buildcraft_1.8_Issues#98]
Using buildcraft-7.2.0-pre6, I drop down to the teens/single digits when one redstone engine is placed in the world.

**Comment by AlexIIL** [comment]
Does this happen when the redstone engine is turned off? or on? or both?

**Comment by AlexIIL** [comment]
Also can you check with the Shift+F3 debug profiler that the main part of the time taken is inside of "root.gameRenderer.level.entities.blockentities"?

**Comment by BBoldt** [comment]
Both. It simply has to be in the world. Doing some more testing now.

**Comment by BBoldt** [comment]
Before placing:
http://i.imgur.com/kI0WsiG.png

After placing:
http://i.imgur.com/68z6Zk2.png

Seems to be lag spikes rather than a constant lag.

**Comment by AlexIIL** [comment]
Its not _directly_ caused by the engine renderer itself then- the percent of time taken up by the renderer is even smaller for after you had placed it. Does the frame rate even out after a period of time? What (in the graph) spikes up as taking the most time?

**Comment by MrGraphic** [comment]
I noticed something very similar when both "markers" and "engines" are placed close to each other - in the same chunk probably forcing additional chunks to constantly updating. Could be the issue ? ( pre7 ).

**Comment by AlexIIL** [comment]
I think I'm going to make engines just a TESR rather than a TESR+BlockModel - engines turning on and off repetitively isn't great for framerate. (Fortunately FastTESR exists so It shouldn't be too slow).
Hopefully that will fix this issue.

**Comment by AlexIIL** [comment]
Yeah, after closer inspection asking minecraft to redraw the mini-chunk every time an engine _receives an update packet_ was a bad idea.
